tools/system_libs.py

```diff
@@ -67,8 +67,12 @@ def build_libc(lib_filename, files, lib_opts):
       commands.append([shared.PYTHON, shared.EMCC, shared.path_from_root('system', 'lib', src), '-o', o] + musl_internal_includes + default_opts + c_opts + lib_opts)
       o_s.append(o)
     run_commands(commands)
-    shared.Building.link(o_s, in_temp(lib_filename))
-    return in_temp(lib_filename)
+    output = in_temp(lib_filename)
+    if shared.get_llvm_target() == shared.WASM_OBJ_TARGET:
+      shared.Building.emar('cr', output, o_s)
```

Comment:
they could be linked using lld, instead of an archive? we specifically only use an archive for libc++ i believe, and this would change that. however, I don't remember all the details of why we do that atm.